src/main/java/com/google/flourbot/entity/EntityModuleImplementation.java

```diff
@@ -1,100 +1,99 @@
 import java.util.concurrent.ExecutionException;
 
 public class EntityModuleImplementation implements EntityModule {
-    
-    private static EntityModuleImplementation instance = null;
-    private final DataStorage datastorage;
 
-    private EntityModuleImplementation() {
-        this.datastorage = new FirebaseDataStorage();
+  private static EntityModuleImplementation instance = null;
+  private final DataStorage datastorage;
+
+  private EntityModuleImplementation() {
+    this.datastorage = new FirebaseDataStorage();
+  }
+
+  public Optional<Macro> getMacro(String userEmail, String macroName)
+      throws InterruptedException, ExecutionException {
+    Optional<QueryDocumentSnapshot> optionalDocument =
+        datastorage.getDocument(userEmail, macroName);
+
+    if (!optionalDocument.isPresent()) {
+      // Handle the case if an empty document is returned.
+      return Optional.empty();
     }
 
-    public Optional<Macro> getMacro(String userEmail, String macroName) throws InterruptedException, ExecutionException {
-        Optional<QueryDocumentSnapshot> optionalDocument = datastorage.getDocument(userEmail, macroName); 
+    // Retrieve value from optional if it's not empty
+    QueryDocumentSnapshot document = optionalDocument.get();
 
-        if (!optionalDocument.isPresent()) {
-            // Handle the case if an empty document is returned.
-            return Optional.empty();
-        } 
-       
-        // Retrieve value from optional if it's not empty 
-        QueryDocumentSnapshot document = optionalDocument.get();
+    Map<String, Object> macroMap = document.getData();
 
-        Map<String, Object> macroMap = document.getData();
+    Map<String, Object> triggerMap = (Map<String, Object>) macroMap.get("trigger");
+    Map<String, Object> actionMap = (Map<String, Object>) macroMap.get("action");
 
-        Map<String, Object> triggerMap = (Map<String, Object>) macroMap.get("trigger");
-        Map<String, Object> actionMap = (Map<String, Object>) macroMap.get("action");
+    Optional<Trigger> optionalTrigger = getTrigger(triggerMap);
+    Optional<Action> optionalAction = getAction(actionMap);
 
-        Optional<Trigger> optionalTrigger = getTrigger(triggerMap);
-        Optional<Action> optionalAction = getAction(actionMap);
+    if (!optionalTrigger.isPresent() || !optionalAction.isPresent()) {
+      return Optional.empty();
+    }
 
-        if (!optionalTrigger.isPresent() || !optionalAction.isPresent()) {
-            return Optional.empty();
-        } 
+    Trigger trigger = optionalTrigger.get();
+    Action action = optionalAction.get();
 
-        Trigger trigger = optionalTrigger.get();
-        Action action = optionalAction.get();
+    String creatorId = (String) macroMap.get("creatorId");
+    Macro macro = new Macro(creatorId, macroName, trigger, action);
 
-        String creatorId = (String) macroMap.get("creatorId");
-        Macro macro = new Macro(creatorId, macroName, trigger, action);
```

Comment:
optionalTrigger.get() as the input works here as well, no need to create a variable if the optional is accessed only once or twice.

Reply:
Done